packages/flutter/lib/src/widgets/scrollbar.dart

```diff
@@ -1869,6 +1867,9 @@ class RawScrollbarState<T extends RawScrollbar> extends State<T> with TickerProv
     if (metrics.axis != _axis) {
       setState(() { _axis = metrics.axis; });
     }
+    if (_isScrollable != notification.metrics.maxScrollExtent > 0) {
```

Comment:
This also is not necessarily true. Some scroll physics allow scrolling when there is no scrollable extent.

Reply:
I made the name too short. It should really be called maxScrollExtentPermitsScrolling; it's only used to detect this one state change and trigger a rebuild.